internal/elasticsearch/logstash/pipeline.go

```diff
@@ -0,0 +1,259 @@
+package logstash
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/elastic/terraform-provider-elasticstack/internal/models"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+var (
+	allSettingsKeys = map[string]schema.ValueType{
+		"pipeline.batch.delay":    schema.TypeInt,
+		"pipeline.batch.size":     schema.TypeInt,
+		"pipeline.workers":        schema.TypeInt,
+		"queue.checkpoint.writes": schema.TypeInt,
+		"queue.max_bytes.number":  schema.TypeInt,
+		"queue.max_bytes.units":   schema.TypeString,
+		"queue.type":              schema.TypeString,
+	}
+)
+
+func ResourceLogstashPipeline() *schema.Resource {
+	logstashPipelineSchema := map[string]*schema.Schema{
+		"id": {
+			Description: "Internal identifier of the resource",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"pipeline_id": {
+			Description: "Identifier for the pipeline.",
+			Type:        schema.TypeString,
+			Required:    true,
+			ForceNew:    true,
+		},
+		"description": {
+			Description: "Description of the pipeline.",
+			Type:        schema.TypeString,
+			Optional:    true,
+		},
+		"last_modified": {
+			Description: "Date the pipeline was last updated.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"pipeline": {
+			Description: "Configuration for the pipeline.",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+		"pipeline_metadata": {
+			Description: "Optional metadata about the pipeline.",
+			Type:        schema.TypeMap,
+			Optional:    true,
+			Elem: &schema.Schema{
+				Type:    schema.TypeString,
+				Default: nil,
+			},
+		},
+		// Pipeline Settings
+		"pipeline_batch_delay": {
```

Comment:
There's a bunch more [settings](https://www.elastic.co/guide/en/logstash/8.4/logstash-settings-file.html) we should look to support. It looks like:

```
pipeline.unsafe_shutdown
pipeline.plugin_classloaders
pipeline.ordered
pipeline.ecs_compatibility
queue.max_events
queue.page_capacity
queue.checkout.acks
queue.checkout.retry
queue.drain
```

I'm happy to add these in a follow up if you're short of time, it would be good to have a more complete group of options though.

Reply:
I was just focusing on the ones visible via the UI initially, I've just updated to include those mentioned too though 👍 Have removed defaults for all to reduce plan noise since all have defaults being set on the API side, happy to read these defaults if you think they are required though?